rootx/src/rootx.cxx

```diff
@@ -472,6 +471,26 @@ int main(int argc, char **argv)
          PrintUsage(argv[0]);
          return 1;
       }
+      if (!strncmp(argv[i], "--version", 9) || !strncmp(argv[i], "-version", 8) || !strncmp(argv[i], "--v", 3) || !strncmp(argv[i], "-v", 2)){
+         FILE *file;
+         char root_version[10];
+         
+         file = popen("root-config --version", "r");
```

Comment:
Call `gROOT->GetVersion()` instead?

Reply:
When I am trying to get the version of root using gROOT->GetVersion(), I am getting following errors: 

This is when I used #define gROOT (ROOT::GetROOT()) to get gROOT:
[ 72%] Built target rootcling_stage1
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:475:36: error: no member
      named 'GetROOT' in namespace 'ROOT'
          printf("Root Version: ", gROOT->GetVersion() + "\n");
                                   ^~~~~
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:442:22: note: expanded
      from macro 'gROOT'
#define gROOT (ROOT::GetROOT())
               ~~~~~~^
1 error generated.


I also tried by including TROOT.h but then I got this issue: 
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:476:36: warning: data
      argument not used by format string [-Wformat-extra-args]
          printf("Root Version: ", gROOT->GetVersion());
                 ~~~~~~~~~~~~~~~~  ^
/Users/aliayubkhan/Documents/build/include/TROOT.h:410:15: note: expanded from
      macro 'gROOT'
#define gROOT (ROOT::GetROOT())
              ^
[ 73%] Built target Base
[ 73%] Built target gitcommit
[ 73%] Built target rootnb.exe
[ 73%] Built target JupyROOT
[ 73%] Linking CXX executable ../bin/root
[ 73%] Built target complexDict
[ 73%] Built target Rint
[ 73%] Built target ThreadObjs
[ 73%] Built target root.exe
[ 73%] Built target Thread
[ 73%] Built target G__Imt
[ 73%] Built target G__RIO
Undefined symbols for architecture x86_64:
  "TVersionCheck::TVersionCheck(int)", referenced from:
      _GLOBAL_sub_I_rootx.cxx in rootx.cxx.o
  "ROOT::GetROOT()", referenced from:
      _main in rootx.cxx.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/root] Error 1
make[1]: *** [rootx/CMakeFiles/root.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Built target Imt
[ 75%] Built target RIOObjs
make: *** [all] Error 2

Reply:
Check your use of `printf`...

Reply:
Oh I have copied wrong logs here.  I am getting this error only on invoking gROOT->GetVersion()  

[ 73%] Built target map2Dict
[ 73%] Built target vectorDict
Undefined symbols for architecture x86_64:
  "TVersionCheck::TVersionCheck(int)", referenced from:
      __GLOBAL__sub_I_rootx.cxx in rootx.cxx.o
  "ROOT::GetROOT()", referenced from:
      _main in rootx.cxx.o
[ 73%] Built target unordered_setDict
[ 73%] Built target mapDict
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[ 73%] Built target unordered_multisetDict
make[2]: *** [bin/root] Error 1
make[1]: *** [rootx/CMakeFiles/root.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Built target forward_listDict
[ 73%] Built target dequeDict
[ 73%] Built target New
[ 73%] Built target unordered_multimapDict
make: *** [all] Error 2